### ISSUE TITLE
Dynamic Multi-Selection Updating

### ISSUE THREAD

**Issue by markdibarry** [godotengine/godot-proposals#9330]
### Describe the project you are working on

`Parallax2D` node

### Describe the problem or limitation you are having in your project

I'd like to increase a property shared by all selected nodes by a certain amount in the editor. In order to perform operations on all values of multiple selected nodes of the same type, a helper parent script needs created or you need to update each one by one manually.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Oftentimes a parent node needs created as an editor shortcut to update values of children of certain type. You can currently perform expressions inside a field value, and you can replace multiple node values with the *same* value, but it would be helpful to be able to select multiple nodes and perform addition or multiplication operations on their current value.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Currently, you can supply expressions in a property value box, and it'll calculate it for you.
![image](https://github.com/godotengine/godot-proposals/assets/21325943/7dd60cf1-22f6-4c9e-97d5-34ad71e7d4af)
![image](https://github.com/godotengine/godot-proposals/assets/21325943/f01e0677-5989-43fa-af95-ef2d78f084ac)

However, there is no way to apply an operation to the current value. This is troublesome when you need to update values of multiple nodes at once, because it limits you to only being able to set all nodes to the same value.

It would be beneficial to allow access to the current value by typing
![image](https://github.com/godotengine/godot-proposals/assets/21325943/ab14dccc-80d3-40a5-b3e8-f36f080721c4)
or something similar if a better syntax were offered.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It can, but it's fairly common and would be a much appreciated shortcut.

### Is there a reason why this should be core and not an add-on in the asset library?

Multi-selection of nodes is a core feature of the editor.

**Comment by Calinou** [comment]
This is something I was thinking a while ago, as it's something I've felt a need for quite often.

The main difficulty is making this feature easy to discover, as `value` isn't something you can guess.

Calculators like SpeedCrunch generally use `ans` to refer to the last returned value:

![image](https://github.com/godotengine/godot-proposals/assets/180032/ad3e3503-347a-4f26-acb1-10fdf9565550)

We could probably support several keywords like `value` and `ans`. I don't know if there's an established standard in other GUI apps.

**Comment by markdibarry** [comment]
Yeah, I'm fine with any, but I understand the goal is usually the least amount of letters while still being easily discoverable. Considering that this would probably only be available for numerical properties, almost anything that isn't expected is a viable option.

**Comment by markdibarry** [comment]
After some investigation, this can be a tricky feature to implement. The multi-select update is pretty rigid with it occurring really late (`_set_impl()` in `multi_node_edit.cpp`), and the evaluation of text happens pretty early on (`_evaluate_input_text()` in `editor_spin_slider.cpp`).

**Comment by kitbdev** [comment]
>  I don't know if there's an established standard in other GUI apps.

In [Unity(Numeric field expressions)](https://docs.unity3d.com/2023.2/Documentation/Manual/EditingValueProperties.html), you can just use `+=`, `-=`, `*=`, or `/=` at the start of the expression to modify the current value. There are also other special functions for multi object editing, such as getting the selected index or a random range.
I don't know if it is in any other apps though, I know in blender the properties are only for one object at a time.

Having a variable may be easier to implement though, I'm not sure.
